packages/core/src/components/overlay/overlay.tsx

```diff
@@ -348,8 +349,18 @@ export class Overlay extends React.PureComponent<IOverlayProps, IOverlayState> {
     private handleDocumentClick = (e: MouseEvent) => {
         const { isOpen, onClose } = this.props;
         const eventTarget = e.target as HTMLElement;
+
+        const { openStack } = Overlay;
+        const stackIndex = openStack.indexOf(this);
+
+        const isClickInDescendantOverlay = openStack
+            .slice(stackIndex)
+            .map(findDOMNode)
```

Comment:
If we change the visibility of `Overlay#containerElement` to `public`, then we can avoid this `findDOMNode` call. I would prefer that approach.